### PR TITLE
🌙 Add GCP Dataproc cluster MQL resource

### PR DESCRIPTION
**Pull request by imilchev**
GCP Dataproc clusters:
- `gcp.project.dataproc.clusters{*}`

### CODE REVIEW

File: resources/packs/gcp/gcp.lr [hunk 1]
```diff
@@ -1251,3 +1253,221 @@ private gcp.project.kmsService.keyring.cryptokey.version.externalProtectionLevel
...
+  config gcp.project.dataprocService.cluster.config
+  // Labels associated with the cluster
+  labels map[string]string
+  // Contains cluster daemon metrics such as HDF and YARN stats
+  metrics dict
+  // Cluster status
+  status gcp.project.dataprocService.cluster.status
+  // Previous cluster status
+  statusHistory []gcp.project.dataprocService.cluster.status
+  // Virtual cluster config is used when creating a Dataproc cluster that does not directly control the underlying compute resources
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]: Do you want to add "Whether" to the beginning of this?

File: resources/packs/gcp/gcp.lr [hunk 2]
```diff
@@ -1251,3 +1253,221 @@ private gcp.project.kmsService.keyring.cryptokey.version.externalProtectionLevel
...
+  virtualClusterConfig gcp.project.dataprocService.cluster.virtualClusterConfig
+}
+
+// GCP Dataproc cluster config
+private gcp.project.dataprocService.cluster.config {
+  // Parent resource path
+  parentResourcePath string
+  // Autoscaling configuration for the policy associated with the cluster
+  autoscaling dict
+  // A Cloud Storage bucket used to stage job dependencies, config files, and job driver console output
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Cloud Storage bucket used to stage job dependencies, config files, and job driver console output
```

Or is "A" really needed?

File: resources/packs/gcp/gcp.lr [hunk 3]
```diff
@@ -1251,3 +1253,221 @@ private gcp.project.kmsService.keyring.cryptokey.version.externalProtectionLevel
...
+  internalIpOnly bool
+  // Compute Engine metadata entries
+  metadata map[string]string
+  // Compute Engine network to be used for machine communications
+  networkUri string
+  // Node Group Affinity for sole-tenant clusters
+  nodeGroupAffinity dict
+  // Type of IPv6 access for the cluster
+  privateIpv6GoogleAccess string
+  // Reservation Affinity for consuming Zonal reservations
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Reservation affinity for consuming zonal reservations
```

File: resources/packs/gcp/gcp.lr [hunk 4]
```diff
@@ -1251,3 +1253,221 @@ private gcp.project.kmsService.keyring.cryptokey.version.externalProtectionLevel
...
+  reservationAffinity gcp.project.dataprocService.cluster.config.gceCluster.reservationAffinity
+  // Dataproc service account used by the Dataproc cluster VM instances
+  serviceAccount string
+  // URIs of service account scopes to be included in Compute Engine instances
+  serviceAccountScopes []string
+  // Shielded Instance config for clusters using Compute Engine Shielded VMs
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Shielded instance config for clusters using Compute Engine Shielded VMs
```

File: resources/packs/gcp/gcp.lr [hunk 5]
```diff
@@ -1251,3 +1253,221 @@ private gcp.project.kmsService.keyring.cryptokey.version.externalProtectionLevel
...
+  shieldedInstanceConfig gcp.project.dataprocService.cluster.config.gceCluster.shieldedInstanceConfig
+  // Compute Engine subnetowrk to be used for machine communications
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Compute Engine subnetwork to use for machine communications
```

File: resources/packs/gcp/gcp.lr [hunk 6]
```diff
@@ -1251,3 +1253,221 @@ private gcp.project.kmsService.keyring.cryptokey.version.externalProtectionLevel
...
+  instanceNames []string
+  // List of references to Compute Engine instances
+  instanceReferences []dict
+  // Whether the instance group contains preemptible instances
+  isPreemptible bool
+  // Compute Engine machine type used for cluster instances
+  machineTypeUri string
+  // Config for Compute Engine Instance Group Manager that manages this group
+  managedGroupConfig dict
+  // Minimum CPU platform for the Instance Group
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Minimum CPU platform for the instance group
```

**Review by misterpantz — COMMENTED** [review]
Beautiful help!!

File: resources/packs/gcp/gcp.lr (hunk 1 above)
Comment (imilchev) [reply resources/packs/gcp/gcp.lr]: Not sure it makes sense. This is not a flag but a whole bunch of settings that are only set whenever a virtual cluster is used

Comment (misterpantz) [reply resources/packs/gcp/gcp.lr]: Ah, I see. I was just trying to figure out why you have "is" in there. Now I get it. I recommend just removing the "is" to be consistent:

```suggestion
  // Virtual cluster config used when creating a Dataproc cluster that does not directly control the underlying compute resources
```

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**